.github/workflows/lint-php.yml

```
@@ -35,12 +39,16 @@ jobs:
   lint-php:
     timeout-minutes: 5
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        php-version: ${{ fromJson(inputs.PHP_MATRIX) }}
     env:
       COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+    strategy:
+      matrix:
+        php-version: ${{ inputs.PHP_MATRIX && fromJSON(inputs.PHP_MATRIX) || fromJson('["custom"]') }}
     steps:
+      - name: Deprecation warning
+        if: ${{ inputs.PHP_MATRIX }}
+        run: echo "::warning::The inputs.PHP_MATRIX is deprecated and will be removed in future. Please use inputs.PHP_VERSION with a matrix in your consuming workflow."
```

The GitHub Actions [documentation](https://docs.github.com/en/actions/using-workflows/reusing-workflows#:~:text=A%20workflow%20that%20uses%20another%20workflow%20is%20referred%20to%20as%20a%20%22caller%22%20workflow.) uses the term "caller workflow".
```suggestion
        run: echo "::warning::The PHP_MATRIX input is deprecated and will be removed in the future. Please create a matrix in your caller workflow and pass it to the PHP_VERSION input."
```
Not sure about the "pass to" wording. Any better idea?

---

I guess "caller" and "pass to" is good to go. 👍🏻 We can make it clearer in the documentation itself.